test/js-api/memory/constructor.any.js

```diff
@@ -128,3 +131,37 @@ test(() => {
   const memory = new WebAssembly.Memory(argument, {});
   assert_Memory(memory, { "size": 0 });
 }, "Stray argument");
+
+test(() => {
+  const argument = { "initial": 1 };
+  const memory = new WebAssembly.Memory(argument);
+  assert_Memory(memory, { "size": 1, "index": "u32" });
+}, "Memory with index parameter omitted");
+
+test(() => {
+  const argument = { "initial": 1, "index": "u32" };
+  const memory = new WebAssembly.Memory(argument);
+  assert_Memory(memory, { "size": 1, "index": "u32" });
+}, "Memory with u32 index constructor");
+
+test(() => {
+  const argument = { "initial": 1, "index": "u64" };
+  const memory = new WebAssembly.Memory(argument);
+  assert_Memory(memory, { "size": 1, "index": "u64" });
+}, "Memory with u64 index constructor");
+
+test(() => {
+  const argument = { "initial": 1, "maximum": 2, "index": "u32", shared: true };
+  const memory = new WebAssembly.Memory(argument);
+  assert_Memory(memory, { "size": 1, "index": "u32", shared: true });
+}, "Shared memory with u32 index constructor");
+
+test(() => {
+  const argument = { "initial": 1, "maximum": 2, "index": "u64", shared: true };
```

Comment:
I think `shared: true` also comes from a spec that has yet to land (https://github.com/WebAssembly/threads) so I don't think we should test that here.

Reply:
Fixed